lib/dalli/ring.rb

```diff
@@ -31,8 +31,8 @@ def server_for_key(key)
       if @continuum
         hkey = hash_for(key)
         20.times do |try|
-          entryidx = binary_search(@continuum, hkey)
-          server = @continuum[entryidx].server
+          entry = @continuum.bsearch { |h| h[:value] <= hkey } || @continuum.last
```

Comment:
This `|| @continuum.last` hack is weird - try removing it and seeing which tests fail. I think I messed up my implementation here somehow.